docs/data/date-pickers/custom-layout/custom-layout.md

```diff
@@ -0,0 +1,95 @@
+---
+product: date-pickers
+title: Date and Time pickers - Custom layout
+components: PickersActionBar, PickersLayout
+githubLabel: 'component: pickers'
+packageName: '@mui/x-date-pickers'
+---
+
+# Custom layout
+
+<p class="description">The Date and Time Pickers let you reorganize the layout</p>
+
+## Default layout structure
+
+By default, pickers are made of 4 sub elements present in the following order:
+
+- The **toolbar** displaying the selected date. Can be enforced with `showToolbar` prop.
+- The **content** displaying the current view. Can be a calendar, or a clock.
+- The **tabs** allowing to switch between day and time views in Date Time Pickers.
+- The **action bar** allowing some interactions. Can be added with [`componentsProps.actionBar`](/x/react-date-pickers/custom-components/#action-bar) prop.
+
+By default the `content` and `tabs` are wrapped together in a `contentWrapper` to simplify the layout.
+
+You can [customize those components](/x/react-date-pickers/custom-components/) individually by using `components` and `componentsProps`.
+
+## Orientation
+
+Toggling layout can be achieved by changing `orientation` prop value between `'portrait'` or `'landscape'`.
+
+Here is a demonstration with the 3 main blocks outlined with color borders.
+
+{{"demo": "LayoutBlocks.js", "hideToolbar": true, "bg": "inline"}}
+
+## Layout structure
+
+The layout is structured as follows:
+
+A `<PickersLayoutRoot />` wraps all the sub-components to provide the structure.
+By default it renders a `div` with `display: grid`.
+Such that all sub-components are placed in a 3 by 3 [CSS grid](https://developer.mozilla.org/fr/docs/Web/CSS/CSS_Grid_Layout).
+
+Subcomponents are wrapped inside layout helper, responsible for placing it into the grid.
+Here is an overview of the structure.
+
+```jsx
+<PickersLayoutRoot>
+  {toolbar}
+  <PickersLayoutContentWrapper>
+    {tabs}
+    {content}
+  </PickersLayoutContentWrapper>
+  {actionBar}
+</PickersLayoutRoot>
+```
+
+## CSS customization
+
+To move an element—the easiest way is to override it's wrapper position with [`gridColumn`](https://developer.mozilla.org/en-US/docs/Web/CSS/grid-column) and [`gridRow`](https://developer.mozilla.org/en-US/docs/Web/CSS/grid-row) properties.
+
+In the next example, the action bar is replaced by a list, and placed at the left of the component.
+Placing at the right is achieved by applying `{ gridColumn: 1, gridRow: 2 }` style to it.
+
+{{"demo": "MovingActions.js"}}
```

Comment:
Changing the `gridColumn` and `gridRow` of the action bar on these demos has no effect.

We need to add the respective class to the custom ActionList to make the selector works. 
This is worth mentioning in the docs as well.

`<List className={pickersLayoutClasses.actionBar}>`

Reply:
There's an open question on using the class to get the gridLayout working. In this case, it had no effect, but it could potentially break something. 
Perhaps we should also handle this case on the docs.

Reply:
It's a broader topic, about how to propagate correctly elements

I updated the demo as follow such that users can to customization directly in the docs, and added a warning such that they know about it

```diff
--- a/docs/data/date-pickers/custom-layout/MovingActions.tsx
+++ b/docs/data/date-pickers/custom-layout/MovingActions.tsx
@@ -10,7 +10,7 @@ import { Unstable_StaticNextDatePicker as StaticNextDatePicker } from '@mui/x-da
 import { PickersActionBarProps } from '@mui/x-date-pickers/PickersActionBar';
 
 function ActionList(props: PickersActionBarProps) {
-  const { onAccept, onClear, onCancel, onSetToday } = props;
+  const { onAccept, onClear, onCancel, onSetToday, className } = props;
   const actions = [
     { text: 'Accept', method: onAccept },
     { text: 'Clear', method: onClear },
@@ -18,7 +18,7 @@ function ActionList(props: PickersActionBarProps) {
     { text: 'Today', method: onSetToday },
   ];
   return (
-    <List>
+    <List className={className}>
       {actions.map(({ text, method }) => (
         <ListItem key={text} disablePadding>
```